File: backend/app/Models/Feed.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Support\Facades\Hash;

class Feed extends Model
{
    use HasFactory;

    protected $fillable =[
        'user_id',
        'content'
    ];
    
    /**
     * appends
     *
     * @var array
     */
    protected $appends = ['liked'];
    
    /**
     * user
     *
     * @return BelongsTo
     */
    public function user(): BelongsTo{
        return $this->belongsTo(User::class);
    }
    
    /**
     * likes
     *
     * @return HasMany
     */
    public function likes(): HasMany{
        return $this->hasMany(Like::class);
    }
    
    /**
     * comments
     *
     * @return HasMany
     */
    public function comments(): HasMany{
        return $this->hasMany(Comment::class);
    }
    
    
    /**
     * getLikedAttribute
     *
     * @return bool
     */
    public function getLikedAttribute(): bool{
        return (bool) $this->likes()->where('feed_id',$this->id)->where('user_id',auth()->id())->exists();
    }

    
}

File: flutter/lib/model/Feed.dart
class Feed {
  Feed({
      this.id, 
      this.userId, 
      this.content, 
      this.createdAt, 
      this.updatedAt, 
      this.liked,});

  Feed.fromJson(dynamic json) {
    id = json['id'];
    userId = json['user_id'];
    content = json['content'];
    createdAt = json['created_at'];
    updatedAt = json['updated_at'];
    liked = json['liked'];
  }
  int? id;
  int? userId;
  String? content;
  String? createdAt;
  String? updatedAt;
  bool? liked;

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    map['id'] = id;
    map['user_id'] = userId;
    map['content'] = content;
    map['created_at'] = createdAt;
    map['updated_at'] = updatedAt;
    map['liked'] = liked;
    return map;
  }

}